libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -55,6 +55,9 @@ type
   MsgIdProvider* =
     proc(m: Message): MessageID {.noSideEffect, raises: [Defect], nimcall, gcsafe.}
 
+  SubscriptionValidator* =
+    proc(topic: string): bool {.noSideEffect, raises: [Defect], nimcall, gcsafe.}
```

Comment:
hm, this is not sufficient to implement a validator that limits the number of subscriptions - it should probably be called for both sub and unsub, with a bool.. the question then would be what to do with failed unsub validations , hm

Reply:
well currently it is called indeed for sub and unsubs btw. cos it's invoked in `subscribeTopic` that deals with both cases.

Reply:
If the topic filters do not change during the app life-time we should have no real issue tho I think.
This filter should be used to filter really unknown topics.

Reply:
is triggering this on `unsubscribe` really needed - even if a peer sends us an unknown topic to unsubscribe from, is it really a fault?

Reply:
Well that is true, but mostly for coherence, logging and scoring (negatively).

Reply:
well, it covers the current use case but I'd say more use cases are covered if we specify the exact operation - or indeed, we make it a subscribe-only thing in which case we should add a few more "basic" limits to libp2p itself, including maximum subscription counts - or a score penalty for each subscription (which would be "regained" by using the subscription in a useful way such as gossipping useful message and getting score bonuses)